armi/reactor/blueprints/blockBlueprint.py

```diff
@@ -162,6 +162,12 @@ def construct(
         b.verifyBlockDims()
         b.spatialGrid = spatialGrid
 
+        if b.spatialGrid is None:
+            try:
+                b.autoCreateSpatialGrids()
+            except (ValueError, NotImplementedError) as e:
+                runLog.warning(str(e), single=True)
+                b.spatialGrid = None
```

Comment:
No need to do this. If it raised, we should  be able to assume that the block is unmodified.